src/index.ts

```diff
@@ -0,0 +1,275 @@
+import {
+	getInput,
+	getMultilineInput,
+	info,
+	setFailed,
+	endGroup,
+	startGroup,
+} from "@actions/core";
+import { execSync, exec } from "node:child_process";
+import { existsSync } from "node:fs";
+import * as path from "node:path";
+import * as util from "node:util";
+const execAsync = util.promisify(exec);
+
+const DEFAULT_WRANGLER_VERSION = "3.4.0";
+
+/**
+ * A configuration object that contains all the inputs & immutable state for the action.
+ */
+const config = {
+	WRANGLER_VERSION: getInput("wranglerVersion") || DEFAULT_WRANGLER_VERSION,
+	secrets: getMultilineInput("secrets"),
+	workingDirectory: checkWorkingDirectory(getInput("workingDirectory")),
+	CLOUDFLARE_API_TOKEN: getInput("apiToken"),
+	CLOUDFLARE_ACCOUNT_ID: getInput("accountId"),
+	ENVIRONMENT: getInput("environment"),
+	VARS: getMultilineInput("vars"),
+	COMMANDS: getMultilineInput("commands"),
+} as const;
+
+function getNpxCmd() {
+	return process.env.RUNNER_OS === "Windows" ? "npx.cmd" : "npx";
+}
+
+async function main() {
+	installWrangler();
+	authenticationSetup();
+	await execCommands(getMultilineInput("preCommands"), "Pre");
+	await uploadSecrets();
+	await wranglerCommands();
+	await execCommands(getMultilineInput("postCommands"), "Post");
+}
+
+function checkWorkingDirectory(workingDirectory = ".") {
+	try {
+		const normalizedPath = path.normalize(workingDirectory);
+		if (existsSync(normalizedPath)) {
+			return normalizedPath;
+		} else {
+			setFailed(`🚨 Directory ${workingDirectory} does not exist.`);
+		}
+	} catch (error) {
+		setFailed(
+			`🚨 While checking/creating directory ${workingDirectory} received ${error}`,
+		);
+	}
+}
+
+function installWrangler() {
+	if (config["WRANGLER_VERSION"].startsWith("1")) {
+		setFailed(
+			`🚨 Wrangler v1 is no longer supported by this action. Please use major versions 2 or 3`,
+		);
+	}
+	startGroup("📥 Installing Wrangler");
+	const command = `npm install wrangler@${config["WRANGLER_VERSION"]}`;
+	info(`Running Command: ${command}`);
+	execSync(command, { cwd: config["workingDirectory"], env: process.env });
+	info(`✅ Wrangler installed`);
+	endGroup();
+}
+
+function authenticationSetup() {
+	startGroup("🔐 Authenticating with Cloudflare");
+	try {
+		process.env.CLOUDFLARE_API_TOKEN = config["CLOUDFLARE_API_TOKEN"];
+		process.env.CLOUDFLARE_ACCOUNT_ID = config["CLOUDFLARE_ACCOUNT_ID"];
+		info(`Authentication process initiated with - API Token`);
+	} catch (error) {
```

Comment:
Why?

Reply:
what error could possible be thrown here? the catch will never be reached, even if the credentials weren't provided. it also doesn't really make sense to log that the authentication process was initiated when all we're doing is setting environment variables. this will always be logged and doesn't give the user much information, so it'll just be extra noise.